modules/kinesis/src/it/scala/com/snowplowanalytics/snowplow/enrich/kinesis/EnrichKinesisSpec.scala

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2022-2022 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.snowplow.enrich.kinesis
+
+import scala.concurrent.duration._
+
+import cats.effect.{IO, Resource}
+
+import retry.syntax.all._
+import retry.RetryPolicies
+
+import cats.implicits._
+
+import cats.effect.testing.specs2.CatsIO
+
+import org.testcontainers.containers.{GenericContainer => JGenericContainer}
+
+import org.specs2.mutable.Specification
+import org.specs2.specification.AfterAll
+
+class EnrichKinesisSpec extends Specification with AfterAll with CatsIO {
+
+  override protected val Timeout = 10.minutes
+
+  val retryPolicy =
+    RetryPolicies.limitRetriesByCumulativeDelay(5.minutes, RetryPolicies.capDelay[IO](2.second, RetryPolicies.fullJitter[IO](1.second)))
+
+  def afterAll: Unit = Containers.localstack.stop()
+
+  "enrich-kinesis" should {
+    "initialize with the minimal config" in {
+      Resource.make[IO, JGenericContainer[_]](
+        IO(Containers.enrich(configPath = "config/config.kinesis.minimal.hocon", loggerName = "minimal"))
+      )(
+        e => IO(e.stop())
+      ).use { e =>
+        IO(e.getLogs must contain("Running Enrich"))
+      }
+    }
+
+    "emit the correct number of enriched events and bad rows" in {
+      import count.Helpers._
+
+      val nbGood = 100l
+      val nbBad = 10l
+
+      Resource.make[IO, JGenericContainer[_]](
+        IO(
+          Containers.enrich(
+            configPath = "modules/kinesis/src/it/resources/enrich/count.hocon",
+            loggerName = "count",
+            network = Some(Containers.localstackNetwork)
```

Comment:
Thinking about this `Option[Network]` - maybe we could get rid of the optional and use this network all the time for each enrich container? That would be a detail hidden in the `enrich` method, not leaking to any of a tests. Would that be a problem if we use it for minimal case, even though it doesn't need that? I think it would remove some noise from tests.

Reply:
Here are the constraints:
- It's possible to run just one of tests (e.g. `sbt "project kinesis" "IntegrationTest/testOnly com.snowplowanalytics.snowplow.enrich.kinesis.EnrichKinesisSpec testOnly -- ex minimal"`
- If only `minimal` is executed, we don't want to start `localstack`
- If both `count` and `stop` are executed, we want to share `localstack`

My hack for that was to use `lazy val` and to use localstack's network for `count` and `stop`, but I also don't like that, I'll come up with something better.

On top of that because of the `def afterAll: Unit = Containers.localstack.stop()` `localstack` container starts to get created before when the tests exits.

Reply:
Great news is that we can call `stop()` on a non-running container, that made my life easier.